src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs

```diff
@@ -45,7 +46,14 @@ public CodeActionsHandler(ICodeFixService codeFixService, ICodeRefactoringServic
                 // If we have a codeaction with a single applychangesoperation, we want to send the codeaction with the edits.
                 var operations = await codeAction.GetOperationsAsync(cancellationToken).ConfigureAwait(false);
 
-                var clientSupportsWorkspaceEdits = clientCapabilities?.Workspace?.WorkspaceEdit?.DocumentChanges == true;
+                var clientSupportsWorkspaceEdits = true;
```

Comment:
extension method?

Reply:
I think you can also do.
clientCapabilitiesExtensions.SelectToken("supportsWorkspaceEdits")?.Value<bool>() ?

Reply:
Did the second - this is only used in one place so I'm not sure it needs an extension method.  New handlers should use the actual client capability